### PR TITLE
[SSO] TENDOFFICE-3089: API endpoint to update a client's setting

### PR DESCRIPTION
**Pull request by annguyen-qh**
Need "github.token", "github.username", "github.email" in SystemSettings

### CODE REVIEW

File: storage/src/main/java/com/wut/resources/settings/SettingResource.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	
+	@Override
+	public Data create(WutRequest request) throws MissingParameterException {
+		String customerDomain = request.getCustomer();
+		Boolean wasSucessful = SettingsManager.createCustomerSettings(customerDomain);
+		return MessageData.successOrFailure(wasSucessful);
+	}
+	@Override
+	public Data delete(WutRequest request) throws MissingParameterException {
+		return null;
```
Comment (rpalmite) [inline storage/src/main/java/com/wut/resources/settings/SettingResource.java]: throw a methodnotsupportedexception. you can find this in other methods that dont support delete.

File: storage/src/main/java/com/wut/resources/settings/SettingResource.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		if (settingValue.isEmpty()) {
+			return ErrorMessage.INVALID_SETTING;
+		} else { 
+			return new StringData(settingValue);
+		}
+	}
+
+	@Override
+	public Data update(WutRequest request) throws MissingParameterException {
+		String setting = request.getParameter("id").toString();
```
Comment (rpalmite) [inline storage/src/main/java/com/wut/resources/settings/SettingResource.java]: this updates one setting

File: storage/src/main/java/com/wut/resources/settings/SettingResource.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+		setting = (setting.equals("theme")) ? "git.branch" : setting;
+		String customer = request.getCustomer();
+		Boolean wasSucessful = SettingsManager.updateCustomerSettings(customer, setting, value);
+		return MessageData.successOrFailure(wasSucessful);
+	}
+	
+	@Override
+	public Data create(WutRequest request) throws MissingParameterException {
+		String customerDomain = request.getCustomer();
```
Comment (rpalmite) [inline storage/src/main/java/com/wut/resources/settings/SettingResource.java]: update is for one setting, but create is for all settings. this behavior is weird. we can leave this for now and fix it in the future.

Comment (annguyen-qh) [inline storage/src/main/java/com/wut/resources/settings/SettingResource.java]: 'create' mean initialize customer setting (with default data). 'update' is use for updating specific setting if customer wanna change

File: storage/src/main/java/com/wut/resources/settings/SettingResource.java (hunk 1 above)
Comment (annguyen-qh) [inline storage/src/main/java/com/wut/resources/settings/SettingResource.java]: you mean `return MessageData.NOT_IMPLEMENTED;`. I see we are using this way

File: core/src/main/java/com/wut/support/settings/SettingsManager.java [hunk 4]
```diff
@@ -196,7 +192,19 @@ public static String getSystemSetting(String settingName) {
 	}
 	
 	// TODO rename getClientSetting()
-	public static String getCustomerSettings(String customer, String settingName) {
+	public static String getCustomerSettings(String customer, String settingName, boolean forceReload) {
+		if (forceReload){
+			System.out.println("forceRefresh Settings");
```
Comment (rpalmite) [inline core/src/main/java/com/wut/support/settings/SettingsManager.java]: this should use our internal WutLogger (i think that's the class) library

File: core/src/main/java/com/wut/support/settings/SettingsManager.java [hunk 5]
```diff
@@ -196,7 +192,19 @@ public static String getSystemSetting(String settingName) {
...
+			clientSettings = ClientSettings.loadFromConfig(true);
+		}
+		return getCustomerSettings(customer, settingName);
+	}
+	public static String getCustomerSettings(String customer, String settingName) {		
+		if ((System.currentTimeMillis() - lastUpdated) > (60*60*1000)){
+			System.out.println("refresh Settings");
```
Comment (rpalmite) [inline core/src/main/java/com/wut/support/settings/SettingsManager.java]: this should use our internal WutLogger (i think that's the class) library

File: storage/src/main/java/com/wut/resources/settings/SettingResource.java (hunk 3 above)
Comment (rpalmite) [inline storage/src/main/java/com/wut/resources/settings/SettingResource.java]: ok. but still weird behavior. :) maybe in the future we'll come up with something better for operations on the collection as a whole.

**Comment by annguyen-qh** [comment]
I read all your new comments. Close this pull request because it's out-of-date, please see #13 instead.

**Pull request closed without merging**